hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/scm/pipeline/TestPipelineManagerMXBean.java

```diff
@@ -83,14 +89,46 @@ private void verifyEquals(TabularData actualData,
       assertEquals(2, cds.values().size());
       Iterator<?> it = cds.values().iterator();
       String key = it.next().toString();
-      String value = it.next().toString();
-      long num = Long.parseLong(value);
       assertTrue(expectedData.containsKey(key));
-      assertEquals(expectedData.remove(key).longValue(), num);
+      // Wait before all pipelines becomes stable in their respective states.
+      GenericTestUtils.waitFor(() -> {
+        long actualCountForPipelineState =
+            getActualCountForPipelineState(key, bean);
+        return expectedData.remove(key).longValue() ==
+            actualCountForPipelineState;
+      }, 500, 3000);
     }
     assertTrue(expectedData.isEmpty());
   }
 
+  private long getActualCountForPipelineState(String pipelineState,
+                                              ObjectName bean) {
+    TabularData data = null;
+    try {
+      data = (TabularData) mbs.getAttribute(bean, "PipelineInfo");
+    } catch (MBeanException e) {
+      throw new RuntimeException(e);
+    } catch (AttributeNotFoundException e) {
+      throw new RuntimeException(e);
+    } catch (InstanceNotFoundException e) {
+      throw new RuntimeException(e);
+    } catch (ReflectionException e) {
+      throw new RuntimeException(e);
```

Comment:
These can be collapsed:

```suggestion
    } catch (MBeanException | AttributeNotFoundException |
             InstanceNotFoundException | ReflectionException e) {
      throw new RuntimeException(e);
```